src/libraries/System.Text.RegularExpressions/tests/RegexReductionTests.cs

```diff
@@ -266,8 +266,26 @@ private static int GetMinRequiredLength(Regex r)
         [InlineData("(?:a+){4}", "a{4,}")]
         [InlineData("(?:a{1,2}){4}", "a{4,8}")]
         // Alternation reduction
```

Comment:
Should there be more test cases here involving inline options? And inline options in subexpressions (which I just learned about)?
Presumably `[InlineData("abcd|(?i)abce", "abc[de]")]`  should be false. Basically test that you're checking node options everywhere you need to.

Reply:
Good idea.  Wouldn't be surprised if it turns up some issues, too.